lib/linguist/heuristics.yml

```diff
@@ -82,12 +82,20 @@ disambiguations:
   - language: LTspice Symbol
     pattern: '^SymbolType[ \t]'
   - language: Asymptote
+- extensions: ['.bas']
+  rules:
+  - language: FreeBasic
+    pattern: '^\s*#(define|endif|endmacro|if|ifdef|ifndef|include|lang|macro)\s'
```

Comment:
🤔 Is this really the only differentiator between VBA and FreeBasic? 

I ask as it'll only match lines that have `#` as the first non-space character and from a quick search on samples of FreeBasic at https://www.freebasic.net/wiki/CodeLibrary, many FreeBasic files don't have lines that start with `#` which means it'll fall through to the classifier which will struggle because your sample looks a lot like VBA, hence the failing test.

Also, why didn't you implement the suggestion from @Alhadis at https://github.com/github/linguist/pull/4998#discussion_r544146635 ? (I've "unresolved" it).

Reply:
I commited the suggestion from @Alhadis. Also in FreeBasic, as opposed to VBA, there are: Dim As, Dim Shared, Common, ...

Reply:
ReDim Shared, ReDim As

Reply:
`Declare Operator`  `#lang "FB"`
And the type that VBA does not have: `Short ZString WString`

Reply:
Given these additional comments, I think you probably need to extend this regex. Maybe @Alhadis has some more suggestions.

Also, should this be using a different regex from that one defined for `.bi`?

Reply:
#lang exists in my pattern

Reply:
>Also, should this be using a different regex from that one defined for .bi?
There is no difference between .bi and .bas

Reply:
I could not add new changes, all attempts failed, so I left it as the previous one.

Reply:
Ok. I think this has gone on long enough. We can always change things later if the current heuristic proves to be problematic.